src/app/shared/utils/utils.ts

```diff
@@ -324,9 +324,7 @@ export async function asyncEvery(arr: any[], predicate: Function) {
   return true;
 }
 
-/** A recursive version of Partial, making all properties, included nested ones, optional.
- * Copied from https://stackoverflow.com/a/47914631
- */
-export type RecursivePartial<T> = {
-  [P in keyof T]?: RecursivePartial<T[P]>;
-};
+/** Extract template name from a full router url */
+export function getTemplateNameFromUrl(urlString: string) {
```

Comment:
Again, this won't be required if using routerLink and routerLinkActive. Also if we ever did want to manually extract what template is used we can directly call the router to extract the `templateName` route parameter on routes such as `/template/my_template`
```ts
{
    path: ":templateName",
    component: TemplatePage,
  },
```
https://angular.io/guide/router#accessing-query-parameters-and-fragments

I'm also wondering why we no longer need the RecursivePartial type - seemed like a nice utility to have. Has it been moved or just no longer required?